vignettes/tcplfit2-vignette.Rmd

```diff
@@ -362,3 +368,263 @@ The output table resulting from the previous code chunk is the same format as th
+points(res_up$bmd, 0, pch = "x", col = "green")
+
+# add marker for BMD and its boundaries after `censoring`
+lines(c(res_up2$bmd,res_up2$bmd),c(0,50),col="blue",lwd=2)
+rect(xleft=res_up2$bmdl,ybottom=0,xright=res_up2$bmdu,ytop=50,col=rgb(0,0,1, alpha = .5), border = NA)
+points(res_up2$bmd, 0, pch = "x", col = "blue")
+
+# add the fitting curve
+lines(conc_plot, poly1(ps = c(res_up$a), conc_plot))
+```
+
+**Figure 9**: This plot shows how BMD and its boundaries changed after "censoring". The green line and green "X" mark the BMD before "censoring" and the green column shows the original boundaries. The blue line and blue "X" mark the "censored" BMD and the blue column shows the updated boundaries. The brown dash line represents the maximum concentration in the data, and the violet dash line represents the threshold value set by `bmd_up_bnd`. In this example, BMD was "censored" to the threshold value, and its boundaries were shifted left.
+
+### "Censor" BMD with tcplhit2_core
+`bmd_low_bnd` and `bmd_low_bnd` are arguments present in `tcplhit2_core`. `concRespCore` "inherit" them because it leverages `tcplhit2_core` within the function. That means, if one chooses to use `tcplfit2_core` and `tcplhit2_core` separately, they can do the same "censoring" for BMD using `tcplhit2_core`. The results from these two methods are identical. Some example codes are provided below to show how to get the same result from the lower bound example above with `tcplhit2_core`.
+```{r example with hit core}
+# using the same data, fit curves 
+param <- tcplfit2_core(conc2, resp2, cutoff = cutoff)
+hit_res <- tcplhit2_core(param, conc2, resp2, cutoff = cutoff, onesd = onesd, 
+                         bmd_low_bnd = 0.8)
+```
+
+```{r res-hit core}
+# adding the result from tcplhit2_core to the output table for comparison
+hit_res["Name"]<-  paste("Chlorothalonil", "tcplhit2_core", sep = "-")
+hit_res['Min. Conc.'] <- min(conc2)
+hit_res[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")] <- round(hit_res[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")], 3)
+
+output_low <- rbind(output_low, 
+                    hit_res[1, c('Name', "Min. Conc.", "bmd", "bmdl", "bmdu")])
+DT::datatable(output_low,rownames = FALSE)
+```
+
+### BMD falls between the "threshold value" and the "reference dose"
+If the estimated BMD falls between the minimum or maximum concentration and the threshold bounds set by `bmd_low_bnd` or `bmd_up_bnd`, the result will be returned as-is. 
+
+Codes below take the example from section ["censor" with lower bound](#censorlowerbound) and show what happens if we use a lower threshold bound. Let's assume it is acceptable as long as the estimated BMD is not lower than 40% (two fifths) of the minimum concentration in the data. The estimated BMD 0.302 is lower than the minimum concentration 0.6, but is higher than 40% of 0.6, which is 0.24. In this case, the BMD estimates and its boundaries will be changed. 
```

Comment:
I would note that somewhere here the following: "for demonstration purposes we use the lower bound censoring example, but the same principle applies for the upper bound censoring case".  

You can put it as the last sentence in the previous paragraph or the first/second sentence for this current paragraph.